v2/selective.go

```diff
@@ -103,19 +128,40 @@ func TraverseToFile(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, sele
 // TraverseV1 walks through the proposed dag traversal and writes a carv1 to the provided io.Writer
 func TraverseV1(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, selector ipld.Node, writer io.Writer, opts ...Option) (uint64, error) {
 	opts = append(opts, WithoutIndex())
+	conf := ApplyOptions(opts...)
 	tc := traversalCar{
-		size:     0,
+		size:     conf.DataPayloadSize,
 		ctx:      ctx,
 		root:     root,
 		selector: selector,
 		ls:       ls,
-		opts:     ApplyOptions(opts...),
+		opts:     conf,
 	}
 
-	len, _, err := tc.WriteV1(writer)
+	len, _, err := tc.WriteV1(tc.ctx, conf.SkipOffset, writer)
 	return len, err
 }
 
+// NewCarV1StreamReader creates an io.ReadSeeker that can be used to copy out the carv1 contents of a car.
+func NewCarV1StreamReader(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, selector ipld.Node, opts ...Option) (io.ReadSeeker, error) {
```

Comment:
this should probably suggest "Selective" shouldn't it? to match `NewSelectiveWriter()`. Perhaps:

```suggestion
func NewSelectiveV1Reader(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, selector ipld.Node, opts ...Option) (io.ReadSeeker, error) {
```